### PR TITLE
drivers: intc: plic: fix IRQ on all CPUs

### PR DESCRIPTION
**Pull request by ycsin**
Enable IRQs on all CPUs, updated the shell command to print the number of IRQ hits on each CPU.

```
west build -b qemu_riscv64_smp -p auto -t run zephyr/samples/hello_world/ -- \
  -DCONFIG_SHELL=y \
  -DCONFIG_SYMTAB=y \
  -DCONFIG_PLIC_SHELL=y \
  -DCONFIG_MP_MAX_NUM_CPUS=4
```

```
*** Booting Zephyr OS build v3.7.0-1079-g17f11332a926 ***
Hello World! qemu_riscv64/qemu_virt_riscv64/smp


uart:~$ plic stats get interrupt-controller@c000000 
   IRQ       CPU 0       CPU 1       CPU 2       CPU 3       Total      ISR(ARG)
     0           1           1           0           1           3      z_irq_spurious(0)
    10           1           3         152           0         167      uart_ns16550_isr(0x80009cf8)

uart:~$ plic stats get interrupt-controller@c000000
   IRQ       CPU 0       CPU 1       CPU 2       CPU 3       Total      ISR(ARG)
     0           1           1           0           1           3      z_irq_spurious(0)
    10           1           3         291           0         306      uart_ns16550_isr(0x80009cf8)

uart:~$ plic stats get interrupt-controller@c000000
   IRQ       CPU 0       CPU 1       CPU 2       CPU 3       Total      ISR(ARG)
     0           1           1           0           1           3      z_irq_spurious(0)
    10           1           3         430           0         445      uart_ns16550_isr(0x80009cf8)
```

### CODE REVIEW

File: drivers/interrupt_controller/intc_plic.c [hunk 1]
```diff
@@ -109,9 +109,9 @@ static inline uint32_t get_plic_enabled_size(const struct device *dev)
 	return local_irq_to_reg_index(config->num_irqs) + 1;
 }
 
-static inline uint32_t get_first_context(uint32_t hartid)
+static ALWAYS_INLINE uint32_t get_hart_context(uint32_t hartid, uint32_t nr_contexts, uint32_t context)
```
Comment (ycsin) [inline drivers/interrupt_controller/intc_plic.c:112]: we need a way to configure the `nr_contexts`

Comment (con-pax) [reply drivers/interrupt_controller/intc_plic.c:112]: Hey @ycsin, I am stuck in the middle of trying to bring Zephyr up in supervisor mode, I can try and get around to testing this out tomorrow

Comment (ycsin) [reply drivers/interrupt_controller/intc_plic.c:112]: > Hey @ycsin, I am stuck in the middle of trying to bring Zephyr up in supervisor mode, I can try and get around to testing this out tomorrow

No worries, good luck with the bring-up

**Comment by ycsin** [comment]
Probably will be superseded by either #77828 / #77645, closed for now

**Pull request closed without merging**